### ISSUE TITLE
GraphQl Cart Errors not retreivable

### ISSUE THREAD

**Issue by Hexmage** [magento/magento2#26830]
If the cart has an error, (eg. because of too low stock), you can't retrieve this error through GraphQl. The error is thrown when you try to place an order through GraphQl. But this is to late in the process.

### Preconditions (*)

1. Magento 2.3.4 Open Source

### Steps to reproduce (*)

1. Add a product to your cart through GraphQl
2. Lower stock of product to below the amount in cart or set product out of stock.
3. Retrieve Cart information through GraphQl.

### Expected result (*)

1. The ability to have access to the hasError value of the Cart
2. The ability to have access to the errorInfo value of the CartItem

### Actual result (*)

1. Error information is unavailable through GraphQl.
```graphql
{
  "data": {
    "cart": {
      "prices": {
        "grand_total": {
          "currency": "USD",
          "value": 34
        }
      },
      "id": "Y4jQzWw93pta8ZLNEeBwygwQlQPbsEyB",
      "items": [
        {
          "id": "5",
          "quantity": 1,
          "product": {
            "name": "Joust Duffle Bag",
            "sku": "24-MB01",
            "description": {
              "html": ""
            }
          }
        }
      ]
    }
  }
}
```
### Additional info
If try to **View and Edit Cart** from Storefront, error messages appear about _Product out of stock_
![out_of_stock](https://user-images.githubusercontent.com/51680745/74337917-e67d3b00-4da9-11ea-96d4-5530e430b3c1.png)

**Comment by Hexmage** [comment]
Still the case in 2.4

**Comment by engcom-Bravo** [comment]
Hello @Hexmage 

This is indeed an issue because when the Cart query is performed in GraphQL, there is no error 
```graphql
{
  "data": {
    "cart": {
      "prices": {
        "grand_total": {
          "currency": "USD",
          "value": 34
        }
      },
      "id": "Y4jQzWw93pta8ZLNEeBwygwQlQPbsEyB",
      "items": [
        {
          "id": "5",
          "quantity": 1,
          "product": {
            "name": "Joust Duffle Bag",
            "sku": "24-MB01",
            "description": {
              "html": ""
            }
          }
        }
      ]
    }
  }
}
```

But if try to **View and Edit Cart** from Storefront, error messages appear about _Product out of stock_
![out_of_stock](https://user-images.githubusercontent.com/51680745/74337917-e67d3b00-4da9-11ea-96d4-5530e430b3c1.png)

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Bravo 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-31440` were created

**Issue Available**: @engcom-Bravo, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by Hexmage** [comment]
@engcom-Bravo I have confirmed that the Error exists on the cartItem object that graphQl uses.

**Comment by lenaorobei** [comment]
This is related to the https://github.com/magento/magento2/issues/26683, please see comment https://github.com/magento/magento2/issues/26683#issuecomment-595290710

**Comment by lenaorobei** [comment]
Should be fixed in the scope of https://github.com/magento/magento2/pull/27015

**Comment by magento-engcom-team** [comment]
:x: Something went wrong. Cannot update Jira issue (https://jira.corp.magento.com/browse/MC-31440).

**Comment by magento-engcom-team** [comment]
:x: Something went wrong. Cannot update Jira issue (https://jira.corp.magento.com/browse/MC-31440).

**Comment by magento-engcom-team** [comment]
:x: Something went wrong. Cannot update Jira issue (https://jira.corp.magento.com/browse/MC-31440).

**Comment by magento-engcom-team** [comment]
:x: Something went wrong. Cannot update Jira issue (https://jira.corp.magento.com/browse/MC-31440).

**Comment by magento-engcom-team** [comment]
:x: Something went wrong. Cannot update Jira issue (https://jira.corp.magento.com/browse/MC-31440).

**Comment by magento-engcom-team** [comment]
:x: Something went wrong. Cannot update Jira issue (https://jira.corp.magento.com/browse/MC-31440).

**Comment by sasha19957099** [comment]
> Should be fixed in the scope of #27015
Can this issue be closed, it already seems to be fixed. Here is my results:

<img width="1216" alt="add_to_cart" src="https://user-images.githubusercontent.com/28062227/85064764-b9f08a00-b1b4-11ea-8d6f-7dd119beb62c.png">
![product_quantity](https://user-images.githubusercontent.com/28062227/85064835-d42a6800-b1b4-11ea-8ab3-5795ddad2f76.png)
<img width="1360" alt="retrieve a cart" src="https://user-images.githubusercontent.com/28062227/85064781-bf4dd480-b1b4-11ea-8737-6fa186c6033a.png">

**Comment by danielrenaud** [comment]
Fixed in https://github.com/magento/magento2/pull/27015

**Comment by DrewML** [comment]
@danielrenaud Are you sure this issue is fixed? The OP mentioned:

> The error is thrown when you try to place an order through GraphQl. But this is to late in the process.

> Steps to reproduce (*)
> Add a product to your cart through GraphQl
> Lower stock of product to below the amount in cart or set product out of stock.
> Retrieve Cart information through GraphQl.

Possible I'm reading it wrong, but it _looks_ like #27015 would only make this information available when you're adding an item _after_ it goes out of stock.

**Comment by danielrenaud** [comment]
@DrewML that is the case that @sasha19957099 demonstrated is now working as expected. The fix in #27015 fixed the CartItems resolver to handle errors (like product out of stock) in a better way.
